cmd/sonobuoy/app/e2e.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2018 Heptio Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"compress/gzip"
+	"fmt"
+	"os"
+
+	"github.com/heptio/sonobuoy/pkg/client"
+	"github.com/heptio/sonobuoy/pkg/errlog"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	show        string
+	rerunFailed bool
+)
+
+func init() {
+	cmd := &cobra.Command{
+		Use:   "e2e",
+		Short: "inspect e2e results",
+		Run:   e2es,
+		Args:  cobra.ExactArgs(1),
+	}
+	RootCmd.AddCommand(cmd)
+	AddKubeconfigFlag(&runFlags.kubecfg, cmd)
+	cmd.PersistentFlags().StringVar(&show, "show", "failed", "Defines which tests to show, options are [passed, failed (default) or all]. Cannot be combined with --rerun-failed.")
+	cmd.PersistentFlags().BoolVar(&rerunFailed, "rerun-failed", false, "Rerun the failed tests reported by the archive. The --show flag will be ignored.")
+}
+
```

Comment:
One thing that isn't clear to me and we should probably outline in a comment block is the extension model for the cli & plugins.  e2e is a special case right now but in the future we should make it pluggable.

Reply:
Pluggable meaning other people can write sonobuoy command line extensions for their own plugins?